x/incentives/keeper/distribute_test.go

```diff
@@ -92,6 +96,201 @@ func (suite *KeeperTestSuite) TestDistribute() {
+	defaultGaugeStartTime := suite.Ctx.BlockTime()
+
+	incentivesParams := suite.App.IncentivesKeeper.GetParams(suite.Ctx).DistrEpochIdentifier
+	currentEpoch := suite.App.EpochsKeeper.GetEpochInfo(suite.Ctx, incentivesParams)
+
+	tests := map[string]struct {
+		// setup
+		numPools           int
+		tokensToAddToGauge sdk.Coins
+		gaugeStartTime     time.Time
+		gaugeCoins         sdk.Coins
+		poolType           poolmanagertypes.PoolType
+		lockExist          bool
+
+		// expected
+		expectErr             bool
+		expectedDistributions sdk.Coins
+	}{
+		"valid case: one poolId and gaugeId": {
+			numPools:              1,
+			gaugeStartTime:        defaultGaugeStartTime,
+			poolType:              poolmanagertypes.Concentrated,
+			gaugeCoins:            sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000))),
+			expectedDistributions: sdk.NewCoins(fiveKRewardCoins),
+			expectErr:             false,
+		},
+		"valid case: gauge with multiple coins": {
+			numPools:              1,
+			gaugeStartTime:        defaultGaugeStartTime,
+			poolType:              poolmanagertypes.Concentrated,
+			gaugeCoins:            sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000)), sdk.NewCoin(appParams.BaseCoinUnit, sdk.NewInt(5000))),
+			expectedDistributions: sdk.NewCoins(fiveKRewardCoins, fiveKRewardCoinsUosmo),
+			expectErr:             false,
+		},
+		"valid case: multiple gaugeId and poolId": {
+			numPools:              3,
+			gaugeStartTime:        defaultGaugeStartTime,
+			poolType:              poolmanagertypes.Concentrated,
+			gaugeCoins:            sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000))),
+			expectedDistributions: sdk.NewCoins(fifteenKRewardCoins),
+			expectErr:             false,
+		},
+		"valid case: attempt to create balance pool": {
+			numPools:              1,
+			poolType:              poolmanagertypes.Balancer,
+			gaugeCoins:            sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000))),
+			gaugeStartTime:        defaultGaugeStartTime,
+			expectedDistributions: sdk.NewCoins(),
+			expectErr:             false, // still a valid case we just donot update the CL incentive parameters
+		},
+		"valid case: lock exist but pool doesnot": {
+			numPools:              0,
+			poolType:              poolmanagertypes.Balancer,
+			gaugeCoins:            sdk.NewCoins(),
+			gaugeStartTime:        defaultGaugeStartTime,
+			expectedDistributions: sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(3000))),
+			lockExist:             true,
+			expectErr:             false, // we do not expect error because we run the gauge distribution to lock logic
+		},
+		"invalid case: attempt to createIncentiveRecord with starttime < currentBlockTime": {
+			numPools:       1,
+			poolType:       poolmanagertypes.Concentrated,
+			gaugeCoins:     sdk.NewCoins(sdk.NewCoin(defaultRewardDenom, sdk.NewInt(5000))),
+			gaugeStartTime: defaultGaugeStartTime.Add(-5 * time.Hour),
+			expectErr:      true,
```

Comment:
Since we are taking a boolean error, did you confirm that the error we are getting in the test case is the one you are expecting?

Reply:
yup i confirmed it before using bool